py_neuromodulation/nm_analysis.py

```diff
@@ -921,31 +922,28 @@ def set_score(
                         take_mean=True,
                     )
 
-                if read_bay_opt_params is True:
+                if read_bay_opt_params:
                     # transform dict into keys for json saving
                     dict_to_save = transform_list_of_dicts_into_dict_of_lists(
                         obj_read["best_bay_opt_params"]
                     )
                     set_score(
                         key_set="bay_opt_best_params",
-                        key_get=None,
                         take_mean=False,
                         val=dict_to_save,
                     )
 
-                if read_mrmr is True:
+                if read_mrmr:
                     # transform dict into keys for json saving
 
                     set_score(
                         key_set="mrmr_select",
-                        key_get=None,
                         take_mean=False,
                         val=obj_read["mrmr_select"],
                     )
-                if model_save is True:
+                if model_save:
```

Comment:
Did mypy complain about those checks? Somehow those checks were for me more interpretable that I directly know `if x is True` then x needs to be boolean. Otherwise x could be any datatype..

Reply:
Oh, I see what was the point of it. Well, this is probably one of those issues where personal preference and coding style makes a huge difference. But I don't think we should waste a lot of time on it so I'll explain my view and you make the call:

- In terms of Python style guide, indeed both the official documentation and the PEP8 style guide recommend against comparisons with True:
    - https://docs.python.org/2/library/stdtypes.html#truth-value-testing
    - https://peps.python.org/pep-0008/
- Is correct that `is True` checks will fail if the variable passed is not a boolean, but we have other levels to account for that:
    - We have type hinting all around the project
    - We have type checking implemented statically and in the close future also dynamically with Pydantic
    - Variable names usually should be enough to tell if a value is a boolean or not. This again comes down to style and personal preference, but when it comes to naming variables changing `model_save` for `save_model` or `saving_enabled` goes a long way.

On the other hand:
- It's pretty much harmless and if it helps you navigate your own project, that's a good reason to keep it. The `is True` part can always be replaced with a single Regex project-wide replacement (what I did basically xD)

Sorry for the long text, it's really not that big a deal xD I'm just that much of a nerd about this stuff i guess.

Reply:
Thanks for the thorough explanation! Pep-0008 actually has a section about boolean comparison:

> Don’t compare boolean values to True or False using ==:
> # Correct:
> if greeting:
> # Wrong:
> if greeting == True:
> Worse:
> 
> # Wrong:
> if greeting is True:

So they're basically arguing against the `is True`. But I think it's a minor point, and I am fine either way, especially if we have type hints it's not a big point.